src/realm/obj.cpp

```diff
@@ -1269,73 +1412,139 @@ void Obj::nullify_link(ColKey origin_col_key, ObjKey target_key)
 
     ColumnAttrMask attr = origin_col_key.get_attrs();
     if (attr.test(col_attr_List)) {
-        Lst<ObjKey> link_list(*this, origin_col_key);
-        size_t ndx = link_list.find_first(target_key);
+        if (origin_col_key.get_type() == col_type_LinkList) {
+            Lst<ObjKey> link_list(*this, origin_col_key);
+            size_t ndx = link_list.find_first(target_link.get_obj_key());
 
-        REALM_ASSERT(ndx != realm::npos); // There has to be one
+            REALM_ASSERT(ndx != realm::npos); // There has to be one
```

Comment:
Seems there is far too much replicated code in this method....

Reply:
Fixed